File: Baekjoon/Swift/TwoPointer/Q3273.swift
//
//  Q3273.swift
//  BOJ_PS
//
//  Created by 김태훈 on 2021/06/19.
//  두 수의 합 / 투 포인터, 정렬

let n = Int(readLine()!)!
var nums = readLine()!.split(separator: " ").map { Int(String($0))! }.sorted()
let x = Int(readLine()!)!

func solution() {
    var start = 0
    var end = n - 1
    var sum = 0
    var res = 0
    
    while start != end {
        sum = nums[start] + nums[end]
        if sum == x {
            start += 1
            res += 1
        } else if sum > x {
            end -= 1
        } else {
            start += 1
        }
    }
    print(res)
}
solution()

File: Baekjoon/Python/BOJ_PS/TwoPointer/Q3273.py
# 두 수의 합 / 투포인터
import sys
input = sys.stdin.readline 

n = int(input())
nums = sorted(list(map(int, input().split())))
x = int(input())
count = 0
i = 0
j = n - 1

while i != j:
    sum = nums[i] + nums[j]
    if sum == x:
        count += 1
        i += 1
    elif sum > x:
        j -= 1
    else:
        i += 1

print(count)